Source/Core/UICommon/GameFileCache.cpp

```diff
@@ -165,7 +165,8 @@ bool GameFileCache::UpdateAdditionalMetadata(std::shared_ptr<GameFile>* game_fil
 {
   const bool wii_banner_changed = (*game_file)->WiiBannerChanged();
   const bool custom_banner_changed = (*game_file)->CustomBannerChanged();
-  if (!wii_banner_changed && !custom_banner_changed)
+
+  if (!wii_banner_changed && !custom_banner_changed && !(*game_file)->UpdateCover())
```

Comment:
This isn't thread-safe. You need to create a copy of the GameFile before you can modify `m_cover` safely.